src/components/App/App.tsx

```diff
@@ -1,26 +1,42 @@
-import { Switch, Route } from 'react-router-dom';
+import { useEffect, useState } from 'react'
+import { Switch, Route, Redirect } from 'react-router-dom';
 import './App.css';
 import Splash from '../Splash/Splash'
 import Dashboard from '../Dashboard/Dashboard'
 import GameScreen from '../GameScreen/GameScreen'
 
 
 const App = () => {
+
+  const [userName, setUserName] = useState<string>('')
+  const [userKey, setUserKey] = useState<string>('')
+
+useEffect(() => {
+  const activeUser = localStorage.getItem('chessAdventureName') || ''
+  const activeKey = localStorage.getItem('chessAdventureKey') || ''
+  setUserName(activeUser)
+  setUserKey(activeKey)
+}, [])
+
   return ( 
     <>
       <Switch>
         <Route 
           exact
           path="/"
-          component={Splash}
-        ></Route>
+          render={() => { return <Splash setUserName={setUserName} setUserKey={setUserKey} />}}
+        >
+          {userKey.length && <Redirect to={`/dashboard`} />}
+        </Route>
         <Route
           exact
-          path="/dashboard/:user"
-          render={({match}) => {
-            return <Dashboard user={match.params.user}/>
+          path="/dashboard"
+          render={() => {
+            return <Dashboard user={userName} />
           }}
-        ></Route>
+          >
+          {!userKey.length && <Redirect to={`/`} />}
```

Comment:
If a user doesn't have local storage credentials they are redirected to the log in page